hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/progress/InstanceProgress.java

```diff
@@ -119,6 +125,10 @@ public void updateInstanceForReductionStep(JobInstance theInstance) {
 
 	public void updateInstance(JobInstance theInstance) {
 		updateInstance(theInstance, false);
+		myWarningMessages.remove(null);
+
+		String newWarningMessage = String.join("\\n", myWarningMessages);
```

Comment:
question: do you mean to escape this?  I was suggesting a plain newline.
```suggestion
		String newWarningMessage = String.join("\n", myWarningMessages);
```

Reply:
It was suppose to be a plain newline, nice catch